exercises/calculator.cpp

```diff
@@ -10,3 +10,37 @@
   Multiplication: 8
   Division: 2
 */
+
+#include <iostream>
+
+using namespace std;
+
+void print_operations(int a, int b)
+{
+  cout << "Sum: " << a + b << endl;
+  cout << "Difference: " << a - b << endl;
+  cout << "Multiplication: " << a * b << endl;
+  cout << "Division: ";
+  if (b == 0)
+  {
+    cout << "impossible" << endl;
+  }
+  else
+  {
+    cout << a / b << endl;
+  }
+}
+
+int main()
+{
+
+  int a, b;
+  cout << "Insert first number: ";
+  cin >> a;
+  cout << "Insert second number: ";
+  cin >> b;
```

Comment:
non male, però, potresti creare una funzione che eviti la ridondanza di codice un po' come qui https://github.com/Helias/git-course-exercises/pull/12/files
leggendo attentamente il commento, potresti anche realizzare una funzione che modifichi il valore della variabile int passandola per riferimento come qui https://github.com/Helias/git-course-exercises/pull/34/files#r870753249

Un'altra cosa che potresti fare, solo se hai voglia, è quella di realizzare questa funzione e cambiare anche l'altra con l'uso dei "template" (tipi generici), in modo da poter "accettare" come tipi anche float e double e non solo "int" (così la tua calcolatrice farà anche le operazioni con numeri con la virgola 😁 )

poi penso che potremmo trovare altre cose da aggiungere, ma al fine dell'esercizio potremmo fermarci qui

Reply:
L'idea dei template non è male, in questo modo rendo compatibili anche gli altri tipi di variabili senza codice ridondante. Per il passaggio per riferimento non ci avevo pensato, grazie per la dritta!